packages/editor/src/components/entities-saved-states/index.js

```diff
@@ -147,7 +146,25 @@ export default function EntitiesSavedStates( { close, onSave = identity } ) {
 		}
 	};
 
-	const saveCheckedEntitiesAndActivate = () => {
+	const {
+		additionalPrompt,
+		isDirty: isDirtyContext,
+		onSave,
+		saveEnabled: saveEnabledContext,
+		saveLabel,
+		setIsDirty: setIsDirtyContext,
+	} = useContext( EntitiesSavedStatesContext );
+	const isDirty =
+		isDirtyContext ??
+		dirtyEntityRecords.length - unselectedEntities.length > 0;
+	useEffect( () => {
+		setIsDirtyContext(
+			dirtyEntityRecords.length - unselectedEntities.length > 0
+		);
+	}, [ dirtyEntityRecords, unselectedEntities, setIsDirtyContext ] );
+	const saveEnabled = saveEnabledContext ?? isDirty;
```

Comment:
Consider defining `isSaveEnabled` to avoid the need to alias `saveEnabled` to `saveEnabledContext`.